Functions.ipynb

```diff
@@ -0,0 +1,633 @@
+    "print_kwargs(city=\"New York\", country=\"USA\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, the print_kwargs function takes a variable number of keyword arguments using the **kwargs syntax. The arguments are printed one by one using a for loop and the items() method of the dictionary."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also mix non-keyword and keyword arguments in a function definition"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Variable Scope"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Variables that are defined inside a function are said to have local scope, which means they can only be accessed within that function. Variables that are defined outside a function have global scope, which means they can be accessed anywhere in the code."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "5\n"
+     ]
+    }
+   ],
+   "source": [
+    "def add_numbers(x, y):\n",
+    "    result = x + y\n",
+    "\n",
+    "add_numbers(2, 3)\n",
+    "\n",
+    "# This will cause an error because 'result' is not defined outside the function\n",
+    "print(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Return Values"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A function can also return a value using the return keyword. For example:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "5\n"
+     ]
+    }
+   ],
+   "source": [
+    "def add_numbers(x, y):\n",
+    "    return x + y\n",
+    "\n",
+    "result = add_numbers(2, 3)\n",
+    "print(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, the add_numbers function returns the sum of x and y. We can call this function and store the return value in a variable"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Lambda Functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In Python, lambda functions are anonymous functions that are defined using the lambda keyword. Lambda functions are typically used for short, one-off functions that are not intended to be reused elsewhere in the code."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here is the general syntax for a lambda function:"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
For code that is not meant to be interpreted, you might want to use:
_```python
lambda arguments: expression
```_
Just for the error/warnings not to be part of it when we don't want them to show up.